### ISSUE TITLE
Missing a Gem Dependency (aws-sdb), and 'rudy init' Command Fails to Read Configuration First

### ISSUE THREAD

**Issue by sabat** [solutious/rudy#12]
Hi -- first, thanks for Rudy.

Issue 1: the gem spec is missing a dependency -- you need to require aws-sdb.

Issue 2: the 'rudy init' command doesn't work out of the box. Follow the instructions in Getting Started, using ruby 1.9.1, and you will get:

AWS credentials must be configured to continue.
You can modify these in /Users/sabat/.rudy/config
ERROR IS Rudy::AWS::SDB::NoAccessKey

For whatever reason, 1.9 isn't picking up the configuration unless you add a call to Huxtable's update_config before you do . I'm throwing over a simple patch, although I have a strong suspicion that just calling update_config isn't the optimal solution.

```
diff -ur rudy-0.7.3.orig/bin/rudy rudy-0.7.3/bin/rudy
--- rudy-0.7.3.orig/bin/rudy    2009-05-09 20:50:59.000000000 -0700
+++ rudy-0.7.3/bin/rudy 2009-05-09 20:51:38.000000000 -0700
@@ -125,6 +125,8 @@

   about "Initialize Rudy configuration"
   command :init do |obj|
+
+    Rudy::Huxtable.update_config

     unless File.exists?(Rudy::CONFIG_FILE)
       Rudy::Config.init_config_dir

diff -ur rudy-0.7.3.orig/rudy.gemspec rudy-0.7.3/rudy.gemspec
--- rudy-0.7.3.orig/rudy.gemspec    2009-05-09 20:50:59.000000000 -0700
+++ rudy-0.7.3/rudy.gemspec 2009-05-09 13:35:41.000000000 -0700
@@ -1,6 +1,6 @@
 @spec = Gem::Specification.new do |s|
    s.name = "rudy"
-  s.rubyforge_project = 'rudy'
+   s.rubyforge_project = 'rudy'
    s.version = "0.7.3"
    s.summary = "Rudy: Not your grandparents' EC2 deployment tool."
    s.description = s.summary
@@ -23,6 +23,7 @@
   s.add_dependency 'json'
   s.add_dependency 'amazon-ec2', '>= 0.3.8'  # Region fix
   s.add_dependency 'aws-s3', '>= 0.6.1'      # Ruby 1.9.1 compatability
+  s.add_dependency 'aws-sdb', '>= 0.6.2'
   s.add_dependency 'net-ssh', '>= 2.0.9'
   s.add_dependency 'net-scp', '>= 1.0.2'
   s.add_dependency 'net-ssh-gateway'
```

[Edit: (2009-05-10) Updated diff formatting -- delano]

**Comment by sabat** [comment]
github screwed up the diff's formatting but you get the idea.

**Comment by delano** [comment]
re: issue 1, thanks for pointing this out. I merged in the aws-sdb library into Rudy (with permission from Tim) but I was still requiring it. It'll be fixed in 0.7.4.

re issue 2, you hit the nail on the head. Rudy used to call Rudy::Huxtable.update_config at require time but I recently changed it to call it at runtime in Rudy::CLI::Base. The init command doesn't subclass CLI::Base so the config wasn't being processed for it. I may revisit the issue later but your fix does the trick and it'll be included in 0.7.4 too.
